gslib/utils/text_util.py

```diff
@@ -263,99 +263,87 @@ def PrintableStr(input_val):
   """
   return input_val
 
+def print_to_fd(*objects, **kwargs):
+  """A Python 2/3 compatible analogue to the print function.
 
-def ttyprint(*objects, **kwargs):
-  """ A Python 2&3 compatible version of the print function.
+  This function writes text to a file descriptor as the
+  builtin print function would, favoring unicode encoding.
 
-  The main purpose of this function is to favor binary output
-  over text output.
-
-  Args are the same as documented for the print function in Python2.
+  Aguments and return values are the same as documented in
+  the Python 2 print function.
   """
-  kw_params = collections.OrderedDict([('sep', ' '),
-                                       ('end', os.linesep),
-                                       ('file', sys.stdout)])
-  for key, value in kwargs.items():
-    if key not in kw_params:
-      raise TypeError(
-        "'{}' is an invalid keyword argument for this function".format(key))
-    kw_params[key] = value
-  sep, end, file = kw_params.values()
-  if six.PY2:
-    if hasattr(file, 'mode') and 'b' not in file.mode and end == os.linesep:
-      end = b'\n'
-    pref_enc = 'utf-8'
-    if isinstance(sep, unicode):
-      sep = sep.encode(pref_enc)
-    if isinstance(end, unicode):
-      end = end.encode(pref_enc)
+  def _get_args(**kwargs):
+    """Validates keyword arguments that would be used in Print
+
+    Valid keyword arguments, mirroring print(), are 'sep',
+    'end', and 'file'. These must be of types string, string,
+    and file / file interface respectively.
+
+    Returns the above kwargs of the above types.
+    """
+    expected_keywords = collections.OrderedDict([
+      ('sep', ' '),
+      ('end', '\n'),
+      ('file', sys.stdout)])
+
+    for key, value in kwargs.items():
+      if key not in expected_keywords:
+        error_msg = (
+          '{} is not a valid keyword argument. '
+          'Please use one of: {}')
+        raise KeyError(
+          error_msg.format(
+            key,
+            ' '.join(expected_keywords.keys())))
+      else:
+        expected_keywords[key] = value
+
+    return expected_keywords.values()
+
+  
+  def _get_normalized_strings(*objects):
```

Comment:
I've got oooone more nit after stepping away from this and re-reading it.  The name on this method doesn't really tell me what's going on; what do you think of naming this something like `_get_byte_strings`?

Also, the docstring is a bit misleading.  The things being passed in aren't necessarily string-like - we're just getting the string representations of them.  Maybe:

```
  """Gets a `bytes` string for each item in a list of printable objects."""
```

Reply:
Both of those sound like good suggestions to me! :+1: Just added a commit with those